Tasks fail to load when period-containing volumeID is defined

Steps to reproduce:

1. Create and run a task with a persistent volume, called `registry.domain`.
2. Wait for the task to be launched
3. Restart Marathon and watch it fail to load tasks:

```
Jul 01 16:51:21 host-3.domain marathon[66449]: [2016-07-01 16:51:21,148] ERROR while loading tasks (akka.actor.OneForOneStrategy:marathon-akka.actor.default-dispatcher-16)
Jul 01 16:51:21 host-3.domain marathon[66449]: java.lang.IllegalStateException: while loading tasks
Jul 01 16:51:21 host-3.domain marathon[66449]:         at mesosphere.marathon.core.task.tracker.impl.TaskTrackerActor$$anonfun$initializing$1.applyOrElse(TaskTrackerActor.scala:102)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at akka.actor.Actor$class.aroundReceive(Actor.scala:465)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at mesosphere.marathon.core.task.tracker.impl.TaskTrackerActor.aroundReceive(TaskTrackerActor.scala:63)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at akka.actor.ActorCell.receiveMessage(ActorCell.scala:516)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at akka.actor.ActorCell.invoke(ActorCell.scala:487)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:254)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at akka.dispatch.Mailbox.run(Mailbox.scala:221)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at akka.dispatch.Mailbox.exec(Mailbox.scala:231)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.pollAndExecAll(ForkJoinPool.java:1253)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1346)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Jul 01 16:51:21 host-3.domain marathon[66449]: Caused by: mesosphere.marathon.SerializationFailedException: registry.domain#storage#8e1f0af7-3fdd-11e6-a2ab-2687a99fcff1 is no valid volumeId
Jul 01 16:51:21 host-3.domain marathon[66449]:         at mesosphere.marathon.core.task.tracker.impl.ReservationSerializer$$anonfun$1.apply(TaskSerializer.scala:214)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at mesosphere.marathon.core.task.tracker.impl.ReservationSerializer$$anonfun$1.apply(TaskSerializer.scala:212)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.Iterator$class.foreach(Iterator.scala:742)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.AbstractIterator.foreach(Iterator.scala:1194)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.TraversableLike$class.map(TraversableLike.scala:245)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.AbstractTraversable.map(Traversable.scala:104)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at mesosphere.marathon.core.task.tracker.impl.ReservationSerializer$.fromProto(TaskSerializer.scala:212)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at mesosphere.marathon.core.task.tracker.impl.TaskSerializer$.reservation$1(TaskSerializer.scala:41)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at mesosphere.marathon.core.task.tracker.impl.TaskSerializer$.fromProto(TaskSerializer.scala:73)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at mesosphere.marathon.core.task.tracker.impl.TaskLoaderImpl$$anonfun$loadTasks$2$$anonfun$apply$4$$anonfun$1.apply(TaskLoaderImpl.scala:24)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at mesosphere.marathon.core.task.tracker.impl.TaskLoaderImpl$$anonfun$loadTasks$2$$anonfun$apply$4$$anonfun$1.apply(TaskLoaderImpl.scala:24)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.TraversableLike$class.map(TraversableLike.scala:245)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.collection.AbstractTraversable.map(Traversable.scala:104)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at mesosphere.marathon.core.task.tracker.impl.TaskLoaderImpl$$anonfun$loadTasks$2$$anonfun$apply$4.apply(TaskLoaderImpl.scala:24)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at mesosphere.marathon.core.task.tracker.impl.TaskLoaderImpl$$anonfun$loadTasks$2$$anonfun$apply$4.apply(TaskLoaderImpl.scala:21)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.util.Success$$anonfun$map$1.apply(Try.scala:237)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.util.Try$.apply(Try.scala:192)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.util.Success.map(Try.scala:237)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
Jul 01 16:51:21 host-3.domain marathon[66449]:         at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
```

Marathon version 1.1.1

--- a/src/main/scala/mesosphere/marathon/core/task/Task.scala
+++ b/src/main/scala/mesosphere/marathon/core/task/Task.scala
@@ -415,7 +415,7 @@ object Task {
   object LocalVolumeId {
     private val uuidGenerator = Generators.timeBasedGenerator(EthernetAddress.fromInterface())
     private val delimiter = "#"
-    private val LocalVolumeEncoderRE = s"^([^.]+)[$delimiter]([^.]+)[$delimiter]([^.]+)$$".r
+    private val LocalVolumeEncoderRE = s"^([^$delimiter]+)[$delimiter]([^$delimiter]+)[$delimiter]([^$delimiter]+)$$".r
 
     def apply(runSpecId: PathId, volume: PersistentVolume): LocalVolumeId =
       LocalVolumeId(runSpecId, volume.containerPath, uuidGenerator.generate().toString)

--- a/src/test/scala/mesosphere/marathon/core/task/TaskTest.scala
+++ b/src/test/scala/mesosphere/marathon/core/task/TaskTest.scala
@@ -1,6 +1,7 @@
 package mesosphere.marathon.core.task
 
 import mesosphere.marathon.MarathonTestHelper
+import mesosphere.marathon.core.task.Task.LocalVolumeId
 import mesosphere.marathon.state.{ AppDefinition, IpAddress, PathId }
 import mesosphere.marathon.test.Mockito
 import org.apache.mesos.{ Protos => MesosProtos }
@@ -120,6 +121,13 @@ class TaskTest extends FunSuite with Mockito with GivenWhenThen with Matchers {
   test("ipAddresses returns one IP for MarathonTask instances with one IP and multiple NetworkInfo") {
     val f = new Fixture
     f.taskWithMultipleNetworksAndOneIp.launched.value.ipAddresses.value should equal(Seq(f.ipAddress1))
+  }
+
+  test("VolumeId should be parsable, even if the task contains a dot in the appId") {
+    val volumeIdString = "registry.domain#storage#8e1f0af7-3fdd-11e6-a2ab-2687a99fcff1"
+    val volumeId = LocalVolumeId.unapply(volumeIdString)
+    volumeId should not be None
+    volumeId should be (Some(LocalVolumeId(PathId.fromSafePath("registry.domain"), "storage", "8e1f0af7-3fdd-11e6-a2ab-2687a99fcff1")))
   }
 
 }

